parser.py

```diff
@@ -589,18 +609,47 @@ def parse_direct_messages(data_folder, username, users, user_id_url_template, dm
           f"({num_written_messages} total messages) to {num_written_files} markdown files\n")
 
 
+def init_paths():
+    dir_archive             = '.'                                                         # used to be `input_folder`
+    dir_input_data          = os.path.join(dir_archive,       'data')                     # used to be `data_folder`
+    dir_input_media         = find_dir_input_media(dir_input_data)                        # used to be `archive_media_folder`
+    dir_output_media        = os.path.join(dir_archive,       'media')                    # used to be `output_media_folder_name`
+    file_output_html        = os.path.join(dir_archive,       'TweetArchive.html')        # used to be `output_html_filename`
+    file_output_following   = os.path.join(dir_archive,       'following.txt')            # used to be `output_following_filename`
+    file_output_followers   = os.path.join(dir_archive,       'followers.txt')            # used to be `output_followers_filename`
+    file_template_dm_output = os.path.join(dir_archive,       'DMs-Archive-{}.md')        # used to be `dm_output_filename_template`
+    file_account_js         = os.path.join(dir_input_data,    'account.js')               # used to be `account_js_filename`
+    file_download_log       = os.path.join(dir_output_media,  'download_log.txt')         # used to be `log_path`
+    file_tweet_icon         = os.path.join(dir_output_media,  'tweet.ico')                # used to be `tweet_icon_path`
+    files_input_tweets      = find_files_input_tweets(dir_input_data)                     # used to be `input_filenames`
+
+    return PathConfig(
+        dir_archive                = dir_archive,
+        dir_input_data             = dir_input_data,
+        dir_input_media            = dir_input_media,
+        dir_output_media           = dir_output_media,
+        file_output_html           = file_output_html,
+        file_output_following      = file_output_following,
+        file_output_followers      = file_output_followers,
+        file_template_dm_output    = file_template_dm_output,
+        file_account_js            = file_account_js,
+        file_download_log          = file_download_log,
+        file_tweet_icon            = file_tweet_icon,
+        files_input_tweets         = files_input_tweets,
+    )
+
+
 def main():
+    paths = init_paths()
```

Comment:
Can we just do:

```
paths = PathConfig(
    dir_archive = '.',
    ...
)
```

and lose `init_paths()` ?

Reply:
I'd love to, but the initializations depend on each other, e.g. `file_download_log = os.path.join(dir_output_media, 'download_log.txt')` needs `dir_output_media` to be initialized. I don't think this is possible inside the arg list of an init method. Same problem for a dict literal.

Also, it looks very verbose due to the structure
```
return PathConfig(
a = a,
b = b,
c = c, ...
```
and I know that `return PathConfig(a, b, c...` would also work, but depends on the order of lines, and would break silently if order gets out of sync).

Reply:
Maybe with an old-style class is easiest: (untested)

```
class PathConfig:
    """Some sensible text here using words that humans can read."""
    def __init__(dir_archive, dir_output):
        self.dir_input_data          = os.path.join(dir_archive, 'data')
        self.dir_input_media         = find_dir_input_media(self.dir_input_data)
        self.dir_output_media        = os.path.join(dir_output,          'media')
        self.file_output_following   = os.path.join(dir_archive,         'following.txt')
        self.file_output_followers   = os.path.join(dir_archive,         'followers.txt')
        self.file_template_dm_output = os.path.join(dir_archive,         'DMs-Archive-{}.md')
        self.file_account_js         = os.path.join(self.dir_input_data, 'account.js')
        self.file_download_log       = os.path.join(dir_output_media,    'download_log.txt')
        self.file_tweet_icon         = os.path.join(dir_output_media,    'tweet.ico')
        self.files_input_tweets      = find_files_input_tweets(dir_input_data)
 
def main():
    paths = PathConfig(dir_archive='.', dir_output='.')
```

Reply:
Hmm, that's nice! Has some minor errors, but once I fixed them, it works. I like this much more. Only downside that I can see: it's not *frozen* without `dataclass`.

Reply:
Anyway, I removed the `dataclass` now. I trust that whenever someone tries to mutate `paths` at some random point, it won't get through review :)

Reply:
True but also applies to every other variable that should be const.

```
from __future__ import const
  File "<stdin>", line 1
SyntaxError: future feature const is not defined
```
:(